content/articles/react-drf-shortener/index.md

```diff
@@ -0,0 +1,369 @@
+
+Open the `settings.py` file and modify its contents as shown below:
+
+For the `INSTALLED_APPS` section:
+
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+
+    'api.apps.ApiConfig', #registers our api app
+
+    'rest_framework',
+
+    'corsheaders'
+]
+```
+
+For the `MIDDLEWARE` section:
+
+```python
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+
+     "corsheaders.middleware.CorsMiddleware",
+
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+```
+
+Append the following code to the `settings.py` file:
+
+```python
+CORS_ALLOWED_ORIGINS = [
+    "http://localhost:3000",
+    "http://127.0.0.1:3000",
+]
+```
+
+The above code specifies the clients that communicate with the Django backend.
+
+Open the `urls.py` file in the `urlshortener` folder and modify it as shown below to route all the requests to the `api` app we previously created.
+
+```python
+from django.contrib import admin
+from django.urls import path, include
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('', include('api.urls')),
+]
+```
+
+### Working on the api app
+
+#### Creating the model
+
+Open the `models.py` file in the api folder and modify it as shown below:
+
+```python
+from django.db import models
+
+class urlShortener(models.Model):
+    longurl = models.CharField(max_length=255)
+    shorturl = models.CharField(max_length=10)
+
+    def __str__(self):
+        return self.shorturl
+```
+
+The above code will create a table in the database with two fields:
+
+-   `longurl` - This will store the long URL submitted by the user from the front-end.
+-   `shorturl` - This will store the short URL equivalent of the long URL generated by the server.
+
+Run the commands below to propagate the changes made to the model into the database schema:
+
+```bash
+python manage.py makemigrations
+
+python manage.py migrate
+```
+
+#### Creating the Serializer
+
+In the `app` folder, create a `serializers.py` file and modify it as shown below:
+
+```python
+from rest_framework.serializers import ModelSerializer
+from .models import urlShortener
+
+class urlShortenerSerializer(ModelSerializer):
+    class Meta:
+        model = urlShortener
+        fields = '__all__
+```
+
+The above code will serialize all fields in the `urlShortener` model.
+
+More information about Serializers can be found [here](https://www.django-rest-framework.org/api-guide/serializers/).
+
+#### Working on the views
+
+Open the `views.py` file and modify it as shown below:
```

Comment:
Add 1 - 2 lines of explanation about what views are. It would be helpful for the beginners. Add reference link if possible.